pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -182,80 +188,102 @@ func validateConstraintNotRepeat(path *field.Path, constraints []v1.TopologySpre
 
 // ValidateRequestedToCapacityRatioArgs validates that RequestedToCapacityRatioArgs are correct.
 func ValidateRequestedToCapacityRatioArgs(args config.RequestedToCapacityRatioArgs) error {
-	if err := validateFunctionShape(args.Shape); err != nil {
-		return err
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateFunctionShape(args.Shape, field.NewPath("shape"))...)
+	allErrs = append(allErrs, validateResourcesNoMax(args.Resources, field.NewPath("resources"))...)
+	return allErrs.ToAggregate()
+}
+
+func validateFunctionShape(shape []config.UtilizationShapePoint, p *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(shape) == 0 {
+		allErrs = append(allErrs, field.Required(p, "at least one point must be specified"))
+		return allErrs
 	}
-	if err := validateResourcesNoMax(args.Resources); err != nil {
-		return err
+
+	for i := 1; i < len(shape); i++ {
+		if shape[i-1].Utilization >= shape[i].Utilization {
+			msg := fmt.Sprintf("utilization values must be sorted. Should < %s==%d", p.Index(i).Child("utilization").String(), shape[i].Utilization)
+			allErrs = append(allErrs, field.Invalid(p.Index(i-1).Child("utilization"), shape[i-1].Utilization, msg))
+		}
 	}
-	return nil
+
+	for i, point := range shape {
+		allErrs = append(allErrs, validateUtilizationShapePoint(point, p.Index(i))...)
```

Comment:
any reason to break this into a new function instead of keeping all the logic here? You could still append the errors based on the index right?

Reply:
No special reason, it's just my habit. I think it would be more clearer to create a function to process a single value instead of embedding it in a list process function. But if you think the previous version could be better, I can revert it.